### PR TITLE
More kernels

### PR DESCRIPTION
**Pull request by thomaspinder**
Implement a series of new kernel functions

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

File: gpjax/config.py [hunk 1]
```diff
@@ -48,6 +52,8 @@ def get_defaults() -> ConfigDict:
     transformations.triangular_transform = FillTriangular
 
     # Default parameter transforms
+    transformations.alpha = "positive_transform"
+    transformations.bias = "positive_transform"
```
Comment (Daniel-Dodd) [inline gpjax/config.py:56]: I would be careful here. To me, the naming "bias", gets me thinking of an element of $\mathbb{R}$ not $\mathbb{R}_{+}$.

File: gpjax/gps.py [hunk 2]
```diff
@@ -436,8 +428,8 @@ def predict(test_inputs: Float[Array, "N D"]) -> GaussianDistribution:
             n_test = test_inputs.shape[0]
 
             μt = mean_function(params["mean_function"], t)
-            Ktt = gram(kernel, params["kernel"], t)
-            Kxt = cross_covariance(kernel, params["kernel"], x, t)
+            Ktt = kernel.gram(params["kernel"], t)
+            Kxt = kernel.cross_covariance(params["kernel"], x, t)
```
Comment (Daniel-Dodd) [inline gpjax/gps.py:432]: This looks neat! Good job. :)

File: gpjax/pytree.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+# limitations under the License.
+# ==============================================================================
+
+import abc
+import jax
+
+from typing import Any
+
+
+class Pytree(metaclass=abc.ABCMeta):
```
Comment (Daniel-Dodd) [inline gpjax/pytree.py:22]: This is my bad. But I think we should call this `PyTree` not `Pytree`. Additionally, we should open an issue of putting this in a small library of JAX utilities - would make it easier to maintain `JaxGaussianProcesses`.

File: gpjax/pytree.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
+# Copyright 2022 The JaxLinOp Contributors. All Rights Reserved.
+#
```
Comment (Daniel-Dodd) [inline gpjax/pytree.py:2]: Perhaps we should be thinking of `JaxGaussianProcesses` or `GPJax` contributors.

**Review by Daniel-Dodd — COMMENTED** [review]
Nice PR, this is a major improvement over before.

**Comment by thomaspinder** [comment]
Suggested kernel changes have now been updated.

**Review by Daniel-Dodd — APPROVED** [review]
Nice PR. Thanks for making the changes.

This prompts opening an issue of dropping the `chex` dependancy. Also instead of ` from chex import PRNGKey as PRNGKeyType`, we can use `from jax.random import `KeyArray` for jax.random.PRNGKey typing.

**Comment by thomaspinder** [comment]
> Nice PR. Thanks for making the changes.
> 
> This prompts opening an issue of dropping the `chex` dependancy. Also instead of ` from chex import PRNGKey as PRNGKeyType`, we can use `from jax.random import `KeyArray` for jax.random.PRNGKey typing.

I agree on dropping Chex. I've opened an issue for this.

**Pull request merged**